.github/workflows/e2e-tests.yml

```diff
@@ -54,6 +39,13 @@ jobs:
     - uses: actions/checkout@v4
       with:
         token: ${{ steps.app-token.outputs.token }}
+    - id: filter
+      name: Check for Rust changes
+      uses: dorny/paths-filter@v3
+      with:
+        filters: |
+          rust:
+            - 'src/wasm-lib/**'
```

Comment:
Moving it here for two reasons:
1. it's simpler
2. allows for easier skip for scheduled or times we don't want to check on changes and want to buid as is

Reply:
Is this just personal preference? I'm wondering because I need to do something similar in #5460.